src/table.rs

```diff
@@ -234,54 +343,58 @@ impl ValueTable {
 		log: &Q,
 		mut f: F,
 	) -> Result<(bool, bool)> {
-		let mut buf: [u8; MAX_ENTRY_SIZE] = unsafe { MaybeUninit::uninit().assume_init() };
+		let mut buf = FullEntry::new();
 
 		let mut part = 0;
 		let mut compressed = false;
 		let entry_size = self.entry_size as usize;
 		loop {
-			let buf = if log.value(self.id, index, &mut buf) {
-				&buf
+			let buf = if log.value(self.id, index, &mut buf.1) {
```

Comment:
There's a lot of `buf.1` in the code now, which makes it a bit harder to read. Looks like `buf.as_mut()` should work here because of `AsMut`. Let's use it instead.